storage.go

```diff
@@ -11,6 +12,53 @@ import (
 	. "github.com/aos-dev/go-storage/v3/types"
 )
 
+func (s *Storage) commitAppend(ctx context.Context, o *Object, opt pairStorageCommitAppend) (err error) {
+	if !o.Mode.IsAppend() {
+		err = fmt.Errorf("object not appendable")
+		return
+	}
+
+	rp := o.GetID()
+
+	offset, ok := o.GetAppendOffset()
+	if !ok {
+		err = fmt.Errorf("append offset is not set")
+		return
+	}
+
+	sessionId := GetObjectMetadata(o).UploadSessionID
+
+	cursor := &files.UploadSessionCursor{
+		SessionId: sessionId,
+		Offset:    uint64(offset),
+	}
+
+	input := &files.CommitInfo{
+		Path: rp,
+		Mode: &files.WriteMode{
+			Tagged: dropbox.Tagged{
+				Tag: files.WriteModeAdd,
+			},
+		},
+	}
+
+	finishArg := &files.UploadSessionFinishArg{
+		Cursor: cursor,
+		Commit: input,
+	}
+
+	fileMetadata, err := s.client.UploadSessionFinish(finishArg, nil)
```

Comment:
Check `err != nil` and return directly, leave complicated logic outside if swtich.